src/client/components/homeComponents/HomePosts.jsx

```diff
@@ -0,0 +1,81 @@
+import { useState, useEffect } from "react";
+
+const URL = process.env.REACT_APP_API_URL;
+const postEndpoint = "/post";
+const postURL = URL + postEndpoint;
+
+const HomePosts = () => {
+  const [postArr, setPostArr] = useState([]);
+  const [search, setSearch] = useState("");
+
+  const handleSearch = e => {
+      setSearch(e.target.value);
+  }
+
+  const searchPosts = (searchTerm, postArr) => {
+    if (!searchTerm) {
+      return postArr;
+    }
+    return postArr.filter(post => post.content.toLowerCase().includes(searchTerm) || post.title.toLowerCase().includes(searchTerm) || post.profile.user.username.toLowerCase().includes(searchTerm))
```

Comment:
This is a very long line of code that does multiple things, it's going to be pretty difficult to debug and read for any future developer (including yourself in 5 weeks time, trust me!)

In general, if something starts to look like this, it's better suited to explicit statements on multiple lines using descriptive, single purpose functions.

```js
function stringIncludesTerm(subject, term) {
  return subject.toLowerCase().includes(term);
}

return postArr.filter(post => {
  const { title, content } = post;
  const { username } = post.profile.user;
  
  if (stringIncludesTerm(title, searchTerm)) {
    return true;
  }
  
  if (stringIncludesTerm(content, searchTerm)) {
    return true;
  }
  
  if (stringIncludesTerm(username, searchTerm)) {
    return true;
  }
  
  return false;
})
```

There are other, cleaner ways you can approach it but this is much easier to see what's happening at a glance. Unless you're working on a high-performance, optimization first project, readability should be favoured.

I know it's more code and doesn't look as cool but you should try to write code as if the person debugging it is a crazy person who knows where you live :)